plugins/woocommerce/client/blocks/assets/js/blocks/product-reviews/inner-blocks/reviews-pagination/edit.tsx

```diff
@@ -0,0 +1,100 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { useSelect } from '@wordpress/data';
+import { PanelBody } from '@wordpress/components';
+import type { BlockEditProps } from '@wordpress/blocks';
+import {
+	InspectorControls,
+	useBlockProps,
+	// @ts-expect-error useInnerBlocksProps is not exported from @wordpress/block-editor
+	useInnerBlocksProps,
+	store as blockEditorStore,
+	Warning,
+} from '@wordpress/block-editor';
+
+/**
+ * Internal dependencies
+ */
+import { ReviewsPaginationArrowControls } from './reviews-pagination-arrow-controls';
+import './editor.scss';
+
+const TEMPLATE = [
+	[ 'woocommerce/product-reviews-pagination-previous' ],
+	[ 'woocommerce/product-reviews-pagination-numbers' ],
+	[ 'woocommerce/product-reviews-pagination-next' ],
+];
+
+type Props = BlockEditProps< { paginationArrow: string | number | undefined } >;
+
+export default function Edit( {
+	attributes: { paginationArrow },
+	setAttributes,
+	clientId,
+}: Props ) {
+	const hasNextPreviousBlocks = useSelect(
+		( select ) => {
+			const { getBlocks } = select( blockEditorStore );
+			const innerBlocks = getBlocks( clientId );
+			/**
+			 * Show the `paginationArrow` control only if a
+			 * Product Reviews Pagination Next or Product Reviews Pagination Previous
+			 * block exists.
+			 */
+			return innerBlocks?.find( ( innerBlock ) => {
```

Comment:
```suggestion
			return innerBlocks?.find( ( innerBlock: { name: string } ) => {
```

I think we can fix the type error here like this.

Reply:
As I commented in https://github.com/woocommerce/woocommerce/pull/57604/files#r2067975246, the reason for the error is the incorrect and outdated type of `@wordpress/block-editor`'s store. It's typed correctly in the recent version, so when we update the type package, the error will be inferred correctly. That's why I leave it as is. But maybe a comment explain that will help?

Reply:
Ah I see, I thought that comment was just referring to the type error for `getBlocks`. Yep OK, I think it's fine to leave it as is then.